File: ios/CoffeeBreak/Models/Boost.swift
import Foundation

/// The 5 power-up visuals (boissons theme), per 03-assets-de-jeu.md.
/// Their effects/powers land with 06-pouvoirs-des-bonus.md.
enum Boost: String, CaseIterable {
    case cafeLatte
    case cafeAEmporter
    case matchaLatte
    case canetteSoda
    case jusOrange

    /// Matches the imageset name in Assets.xcassets/Boosts.
    var assetName: String {
        switch self {
        case .cafeLatte: return "boost-cafe-latte"
        case .cafeAEmporter: return "boost-cafe-a-emporter"
        case .matchaLatte: return "boost-matcha-latte"
        case .canetteSoda: return "boost-canette-soda"
        case .jusOrange: return "boost-jus-orange"
        }
    }
}

File: mobile/lib/models/boost.dart
/// The 5 boost visuals, per 03-assets-de-jeu.md.
enum Boost {
  cafeLatte,
  cafeAEmporter,
  matchaLatte,
  canetteSoda,
  jusOrange;

  String get assetName {
    switch (this) {
      case Boost.cafeLatte:
        return 'assets/boosts/boost-cafe-latte.png';
      case Boost.cafeAEmporter:
        return 'assets/boosts/boost-cafe-a-emporter.png';
      case Boost.matchaLatte:
        return 'assets/boosts/boost-matcha-latte.png';
      case Boost.canetteSoda:
        return 'assets/boosts/boost-canette-soda.png';
      case Boost.jusOrange:
        return 'assets/boosts/boost-jus-orange.png';
    }
  }
}
